lib/solr/query/request/facet.rb

```diff
@@ -28,16 +29,17 @@ def initialize(field:, type:, name: nil, value: nil, filters: [], subfacets: [],
             raise ArgumentError, "Need to specify :lower_bound, :upper_bound, :gap option values for 'range' facet type"
           end
 
-          @field       = field
-          @name        = name || field
-          @type        = type
-          @value       = value
-          @filters     = filters
-          @subfacets   = subfacets
-          @limit       = options[:limit].to_i
-          @gap         = options[:gap]
-          @lower_bound = options[:lower_bound]
-          @upper_bound = options[:upper_bound]
+          @field        = field
+          @name         = name || field
+          @type         = type
+          @value        = value
+          @filters      = filters
+          @subfacets    = subfacets
+          @limit        = options[:limit].to_i
+          @gap          = options[:gap]
+          @lower_bound  = options[:lower_bound]
+          @upper_bound  = options[:upper_bound]
+          @facet_filter = options[:facet_filter]
```

Comment:
confusing naming especially considering we already have: https://github.com/machinio/solrb/pull/25/files#diff-1bebb5c8dec01b166e5591b4813ad214R36
How are these 2 different?

Reply:
The difference is that this `facet_filter` I added will filter all the query response records before faceting:
https://lucene.apache.org/solr/guide/7_2/json-facet-api.html#FilteringFacets

The `filters` here is not actually for filtering, it's being used to generate query facets, the naming in this case is not good.
https://lucene.apache.org/solr/guide/7_2/json-facet-api.html#QueryFacet
https://github.com/machinio/solrb/blob/master/lib/solr/query/request/facet.rb#L57
Since we're still in `0.*` version I think it's worth to change the name of this param to `queries` or something similar.